Spigot-Server-Patches/0371-Async-Chunk-Loading-and-Generation.patch

```diff
@@ -339,7 +344,11 @@ index 000000000..5c77b6e8e
 +        }
 +
 +        public void bumpPriority() {
-+            if (!priority.compareAndSet(Priority.NORMAL.ordinal(), Priority.HIGH.ordinal())) {
++            bumpPriority(Priority.HIGH);
++        }
++
++        public void bumpPriority(Priority newPriority) {
++            if (priority.get() == newPriority.ordinal()) {
```

Comment:
As the code stands now, NO change in priority will occur as priority is not written to.

This needs to be a CAS loop ensuring that a priority update occurs if and only if the new priority is higher than the current.

Something like the following would work:
https://gist.github.com/Spottedleaf/0357a76ba3fcd82b3b283eff8d971963

Reply:
Cheers, I somehow managed to forget to jump back to dealing with the "and set" portion of this properly, also never really considered a concern over that priority field given how this code works, however, I guess given the util designation, we might as well do it properly